### PR TITLE
Various Key/Row/Col + FromStringImpl fixes.

### PR DESCRIPTION
**Pull request by mzhurovich**

### CODE REVIEW

File: Bricks/strings/util.h [hunk 1]
```diff
@@ -65,75 +65,78 @@ constexpr auto HasMemberFromString(int)
...
+  static std::string DoIt(bool b) { return b ? "true" : "false"; }
 };
 
-template <typename DECAYED_T, bool B>
-struct ToStringImpl<DECAYED_T, true, B> {
-  template <typename T>
-  static std::string DoIt(T&& x) {
-    return x.ToString();
-  }
+// Types that implement `ToString()` method.
```
Comment (dkorolev) [inline Bricks/strings/util.h]:
```
... the `ToString` method.
```

File: Storage/container/sfinae.h [hunk 2]
```diff
@@ -81,15 +87,15 @@ using key_accessor_t = typename vs_impl_key_accessor_t<ENTRY>::type;
 #endif  // CURRENT_WINDOWS
 
 template <typename ENTRY>
-typename key_accessor_t<ENTRY>::key_t GetKey(const ENTRY& entry) {
```
Comment (dkorolev) [inline Storage/container/sfinae.h]: Note to self: Reviewed until here.

File: Storage/container/sfinae.h [hunk 3]
```diff
@@ -112,39 +118,47 @@ struct impl_row_accessor_t {};
...
 };
 
 template <typename ENTRY>
 struct impl_row_accessor_t<ENTRY, true> {
-  typedef decltype(std::declval<ENTRY>().row()) row_t;
-  // Can not return a reference to a temporary.
-  static const row_t GetRow(const ENTRY& entry) { return entry.row(); }
-  static void SetRow(ENTRY& entry, CF<row_t> row) { entry.set_row(row); }
+  using cf_row_t = decltype(std::declval<ENTRY>().row());
+  using row_t = current::decay<decltype(std::declval<ENTRY>().row())>;
```
Comment (dkorolev) [inline Storage/container/sfinae.h]: Duplicate decltype intentional?

File: Storage/container/sfinae.h [hunk 4]
```diff
@@ -112,39 +118,47 @@ struct impl_row_accessor_t {};
...
+  static cf_row_t GetRow(const ENTRY& entry) { return entry.row(); }
+  static void SetRow(ENTRY& entry, cf_row_t row) { entry.set_row(row); }
 };
 
 template <typename ROW, typename COL>
 struct impl_row_accessor_t<std::pair<ROW, COL>, false> {
-  typedef ROW row_t;
-  static CF<row_t> GetRow(const std::pair<ROW, COL>& entry) { return entry.first; }
-  static void SetRow(std::pair<ROW, COL>& entry, CF<row_t> row) { entry.first = row; }
+  static_assert(std::is_same<ROW, current::decay<ROW>>::value, "");
```
Comment (dkorolev) [inline Storage/container/sfinae.h]: Same `static_assert` for `COL`?

File: Storage/container/sfinae.h [hunk 5]
```diff
@@ -112,39 +118,47 @@ struct impl_row_accessor_t {};
...
   return row_accessor_t<ENTRY>::GetRow(entry);
 }
 
 template <typename ENTRY>
-using entry_row_t = current::decay<typename row_accessor_t<ENTRY>::row_t>;
+using entry_row_t = typename row_accessor_t<ENTRY>::row_t;
 
 template <typename ENTRY>
-void SetRow(ENTRY& entry, CF<entry_row_t<ENTRY>> row) {
+void SetRow(ENTRY& entry, typename row_accessor_t<ENTRY>::cf_row_t row) {
```
Comment (dkorolev) [inline Storage/container/sfinae.h]: Are you sure this would compile?

File: Storage/container/sfinae.h [hunk 6]
```diff
@@ -163,39 +177,47 @@ struct impl_col_accessor_t {};
 
 template <typename ENTRY>
 struct impl_col_accessor_t<ENTRY, false> {
-  typedef decltype(std::declval<ENTRY>().col) col_t;
-  static CF<col_t> GetCol(const ENTRY& entry) { return entry.col; }
-  static void SetCol(ENTRY& entry, CF<col_t> col) { entry.col = col; }
+  using col_t = decltype(ENTRY::col);
+  using cf_col_t = CF<col_t>;
+
```
Comment (dkorolev) [inline Storage/container/sfinae.h]: `static_assert` (s) here too, same as for the row?

Comment (dkorolev) [inline Storage/container/sfinae.h]: Ah, take it back.

File: Storage/container/sfinae.h [hunk 7]
```diff
@@ -163,39 +177,47 @@ struct impl_col_accessor_t {};
...
+  static cf_col_t GetCol(const ENTRY& entry) { return entry.col(); }
+  static void SetCol(ENTRY& entry, cf_col_t col) { entry.set_col(col); }
 };
 
 template <typename ROW, typename COL>
 struct impl_col_accessor_t<std::pair<ROW, COL>, false> {
-  typedef COL col_t;
-  static CF<col_t> GetCol(const std::pair<ROW, COL>& entry) { return entry.second; }
-  static void SetCol(std::pair<ROW, COL>& entry, CF<col_t> col) { entry.second = col; }
+  static_assert(std::is_same<COL, current::decay<COL>>::value, "");
```
Comment (dkorolev) [inline Storage/container/sfinae.h]: `static_assert` on a `ROW` too here, just in case?

**Comment by dkorolev** [comment]
LGTM :+1:

File: Bricks/strings/util.h (hunk 1 above)
Comment (mzhurovich) [inline Bricks/strings/util.h]: 👍

File: Storage/container/sfinae.h (hunk 3 above)
Comment (mzhurovich) [inline Storage/container/sfinae.h]: Fixed.

File: Storage/container/sfinae.h (hunk 4 above)
Comment (mzhurovich) [inline Storage/container/sfinae.h]: Added.

File: Storage/container/sfinae.h (hunk 5 above)
Comment (mzhurovich) [inline Storage/container/sfinae.h]: Why not?

**Pull request merged**